src/goto-programs/abstract-interpretation/interval_domain.cpp

```diff
@@ -891,13 +901,30 @@ void interval_domaint::transform(
     assume(instruction.guard);
     break;
 
-  case FUNCTION_CALL:
+  case RETURN:
   {
-    const code_function_call2t &code_function_call =
-      to_code_function_call2t(instruction.code);
-    if (!is_nil_expr(code_function_call.ret))
-      havoc_rec(code_function_call.ret);
-    break;
+    // After a return, all function arguments becomes nondet
+    const symbolt *current_function = ns.lookup(instruction.function);
+    type2tc t = migrate_type(current_function->type);
+    const code_type2t &function = to_code_type(t);
+
+    for (size_t i = 0; i < function.arguments.size(); i++)
+    {
+      const type2tc &arg_type = function.arguments[i];
+      const expr2tc arg_symbol =
+        symbol2tc(arg_type, function.argument_names[i]);
+      havoc_rec(arg_symbol);
+    }
+
+    /* The current implementation of the abstract interpreter do not store
+     * the return variable (which would be too tricky anyway). We can deal
+     * with this by constructing a tmp symbol in which we can apply assumptions
+     * later */
+    expr2tc return_var = symbol2tc(
+      function.ret_type, fmt::format("c:{}:ret", instruction.function));
+
+    assign(
+      code_assign2tc(return_var, to_code_return2t(instruction.code).operand));
```

Comment:
Is it guaranteed that the operand and return_var exist? What happens with `void` functions that just `return;`?

Reply:
1. The interval analysis will select the correct domain based on the type (i.e. bitvectors or floats). An unsupported domain won't affect the abstract state at all.
2. Based on the flow, the return variable should always be initialized before being used (as it requires a function call and a return to have happened before). Nevertheless, if a var was not initialized in the domain, we just assume it to be a top (i.e., -inf to inf).

Reply:
Not sure I fully understand. Let's say there is a `return;` in a function returning `void`. What would it mean for the operand and the type of `return_var`? (+/-inf) doesn't really make sense for `void`, does it? Maybe I'm missing something.

Reply:
> Let's say there is a return; in a function returning void.

Since this is neither a bitvector (for wrapped and integer domains) and neither a float/fixed (for real domain)... we will just ignore it at the assign function.

> (+/-inf) doesn't really make sense for void, does it? 

It doesn't, I meant for what happens if some crazy flow tries to use a variable that was not initialized. This should be impossible though (maybe if someone does some magic if longjmp?).

Reply:
Makes sense, thanks!

> (maybe if someone does some magic if longjmp?)

Fortunately not implemented, yet. But that's a valid concern we'll have to address when the time comes.